From 521c96781eb2de517ae5bdfb1601d06c0b6c1ffc Mon Sep 17 00:00:00 2001
From: Tanu Kaskinen <tanu.kaskinen@jollamobile.com>
Date: Wed, 10 Oct 2012 11:07:48 +0300
Subject: [PATCH 2/7] Use /etc/boardname to load a hardware specific
 configuration file.

I think it would be better to add support for /etc/boardname directly
to the code that loads the configuration file, but since the
start-pulseaudio already existed in Mer, it's easier to keep using
that.
---
 src/Makefile.am                  |  2 +-
 src/daemon/pulseaudio.desktop.in |  2 +-
 src/daemon/start-pulseaudio      | 41 ++++++++++++++++++++++++++++++++++++++++
 src/pulse/client.conf.in         |  2 +-
 4 files changed, 44 insertions(+), 3 deletions(-)
 create mode 100755 src/daemon/start-pulseaudio

diff --git a/src/Makefile.am b/src/Makefile.am
index 88db31c..04fdbb5 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -176,7 +176,7 @@ endif
 #       Utility programs          #
 ###################################
 
-bin_SCRIPTS = esdcompat
+bin_SCRIPTS = esdcompat daemon/start-pulseaudio
 
 bin_PROGRAMS += \
 		pacat \
diff --git a/src/daemon/pulseaudio.desktop.in b/src/daemon/pulseaudio.desktop.in
index b75b378..5049953 100644
--- a/src/daemon/pulseaudio.desktop.in
+++ b/src/daemon/pulseaudio.desktop.in
@@ -3,7 +3,7 @@ Version=1.0
 Encoding=UTF-8
 _Name=PulseAudio Sound System
 _Comment=Start the PulseAudio Sound System
-Exec=start-pulseaudio-x11
+Exec=start-pulseaudio
 Terminal=false
 Type=Application
 Categories=
diff --git a/src/daemon/start-pulseaudio b/src/daemon/start-pulseaudio
new file mode 100755
index 0000000..d74735a
--- /dev/null
+++ b/src/daemon/start-pulseaudio
@@ -0,0 +1,41 @@
+#!/bin/sh
+
+# This file is part of PulseAudio.
+#
+# PulseAudio is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# PulseAudio is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with PulseAudio; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307
+# USA.
+
+CONFIG_SELECT=""
+# Boardname is not mandatory to have
+if [ -e /etc/boardname ]; then
+    BOARDNAME_REVISION=`cat /etc/boardname`
+    BOARDNAME=`echo $BOARDNAME_REVISION | sed 's!_[a-zA-Z0-9]*$!!g'`
+
+    if [ -e /etc/pulse/$BOARDNAME_REVISION.pa ]; then
+        # Found configuration specific to board and revision.
+        CONFIG_SELECT="-n --file=/etc/pulse/$BOARDNAME_REVISION.pa"
+    elif [ -e /etc/pulse/$BOARDNAME.pa ]; then
+        # Found configuration specific to board.
+        CONFIG_SELECT="-n --file=/etc/pulse/$BOARDNAME.pa"
+    fi
+fi
+
+if [ -x /usr/bin/start-pulseaudio-x11 ] ; then
+    /usr/bin/start-pulseaudio-x11 $CONFIG_SELECT "$@"
+else
+    set -e
+    [ -z "$PULSE_SERVER" ]
+    /usr/bin/pulseaudio --start $CONFIG_SELECT "$@"
+fi
\ No newline at end of file
diff --git a/src/pulse/client.conf.in b/src/pulse/client.conf.in
index 17753b0..85c8890 100644
--- a/src/pulse/client.conf.in
+++ b/src/pulse/client.conf.in
@@ -25,7 +25,7 @@
 ; default-dbus-server =
 
 ; autospawn = yes
-; daemon-binary = @PA_BINARY@
+daemon-binary = /usr/bin/start-pulseaudio
 ; extra-arguments = --log-target=syslog
 
 ; cookie-file =
-- 
1.8.4

